cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/batch/BatchContext.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.api.batch;
+
+import co.cask.cdap.api.Resources;
+import co.cask.cdap.api.RuntimeContext;
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.data.batch.BatchReadable;
+import co.cask.cdap.api.data.batch.InputFormatProvider;
+import co.cask.cdap.api.data.batch.OutputFormatProvider;
+import co.cask.cdap.api.data.batch.Split;
+import co.cask.cdap.api.dataset.Dataset;
+
+import java.util.List;
+
+/**
+ * Context passed to Batch Sources and Sink.
+ */
+public interface BatchContext extends RuntimeContext, DatasetContext {
+
+  /**
+   * Returns the logical start time of the Batch Job.  Logical start time is the time when this MapReduce
+   * job is supposed to start if this job is started by the scheduler. Otherwise it would be the current time when the
+   * job runs.
+   *
+   * @return Time in milliseconds since epoch time (00:00:00 January 1, 1970 UTC).
+   */
+  long getLogicalStartTime();
+
+  /**
+   */
+  <T> T getHadoopJob();
```

Comment:
i think you can have this return the actual Job. I believe the reason MapReduceContext does this is to avoid bringing in mapreduce as a dependency into api.